src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -97,6 +97,11 @@ internal IOperation GetOperationInternal(SyntaxNode node, CancellationToken canc
         protected abstract IOperation GetOperationCore(SyntaxNode node, CancellationToken cancellationToken);
 
         /// <summary>
+        /// Deep Clone given IOperation
+        /// </summary>
+        internal abstract T CloneOperation<T>(T operation) where T : IOperation;
```

Comment:
Can we avoid a generic virtual method, and generic methods in several classes, by introducing an abstract `CloneOperationCore` method?
```
internal abstract IOperation CloneOperationCore(IOperation op);

internal T CloneOperation<T>(T op) where T : IOperation => (T)CloneOperationCore(op);
```